src/Microsoft.SqlTools.ServiceLayer/SqlProjects/Contracts/Folders/AddFolder.cs

```diff
@@ -3,6 +3,8 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
+#nullable disable
```

Comment:
Added by mistake? Or is it necessary?

Reply:
Intentional.  It's a bit of a funky balance: we want to be able to turn on full nullable checks, but because the JRPC deserializer uses a blank constructor followed by property sets, we can't enforce that "required" properties (like `Path` below) are actually set.  But it's also not meant to be nullable, so I don't want to mark it as `string?` to imply that it's okay if it's null.

The choice then becomes either setting Path to be `string?` and force everything that uses it to handle null cases, or add the `#nullable disable` hint back.

... Or we could probably do something like having a secondary field with proper nullability (or lack thereof) that actually gets used in Tools Service:

```
public string? Path { get; set; } // set by the JRPC deserializer, never called except below
public string GetPath => Path ?? throw new ArgumentNullException(); // used in code, handles the null check for everything that uses it
```

... Or maybe if the JRPC deserializer uses init syntax to set these, they we can use new accessor scopes:

```
public string Path {get; init set; }
```

But that's something I'll have to play with.